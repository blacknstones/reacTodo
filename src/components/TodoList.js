import React, { useEffect, useState } from 'react';
import AddTaskForm from './AddTaskForm';
import TodoCard from './TodoCard';

const TodoList = () => {
  const [tasks, setTasks] = useState([]);

  const onSubmit = newTask => {
    setTasks([...tasks, newTask]);
  };

  const removeTask = id => {
    const newTasks = tasks.filter(el => el.id !== id);
    setTasks(newTasks);
  };

  const editTask = newTask => {
    let newTasks = tasks;
    const index = tasks.findIndex(el => el.id === newTask.id);
    newTasks[index] = newTask;
    setTasks(newTasks);

  };

  useEffect(() => {
    console.log('tasks', tasks);
  }, [tasks]);

  return (
    <main>
      <AddTaskForm onSubmit={data => onSubmit(data)} />
      <ul className='todo'>
        {tasks.map(task => (
          <TodoCard key={task.id} task={task} removeTask={removeTask} editTask={editTask}/>
        ))}
      </ul>
    </main>
  );
};

export default TodoList;
