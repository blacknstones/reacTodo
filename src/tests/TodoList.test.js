import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from '../components/TodoList';

describe('TodoList component', () => {
  test('TodoList renders empty todolist on first load', () => {
    const TodoComponent = renderer.create(<TodoList />);
    let tree = TodoComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
