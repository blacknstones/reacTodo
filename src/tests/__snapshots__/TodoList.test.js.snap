// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component TodoList renders empty todolist on first load 1`] = `
<main>
  <form
    id="form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="input__title"
    >
      Title
    </label>
    <input
      id="input__title"
      name="title"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      htmlFor="input__desc"
    >
      Description
    </label>
    <input
      id="input__desc"
      name="description"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      id="button-add"
      type="submit"
      value="Add"
    />
  </form>
  <ul
    className="todo"
  />
</main>
`;
